src/transformers/pipelines/base.py

```diff
@@ -792,6 +792,12 @@ def __init__(
         self._num_workers = kwargs.pop("num_workers", None)
         self._preprocess_params, self._forward_params, self._postprocess_params = self._sanitize_parameters(**kwargs)
 
+        if self.image_processor is None and self.feature_extractor is not None:
```

Comment:
To ensure backward compatibility

Reply:
You are right, I am so ignorant , sorry. I guess it won't hurt, but potentially leads to some confusion if one abuse the attributes.

One way is to verify if `self.feature_extractor` is an instance of `BaseImageProcessor`